From 84f7498f14dae96a078c3026ecde3712449279fb Mon Sep 17 00:00:00 2001
From: xc-racer99 <xc-racer2@live.ca>
Date: Fri, 26 Aug 2016 18:10:53 -0700
Subject: [PATCH] bt: Fix compilation for GATT/SMP/LE disabled

Lightly tested, but it compiles, I can pair with a computer,
and play audio to it
---
 Android.mk              |  5 +++++
 bta/dm/bta_dm_act.c     |  4 ++++
 btif/src/btif_dm.c      |  4 ++++
 btif/src/btif_storage.c |  2 ++
 stack/l2cap/l2c_csm.c   | 22 ++++++++++++++++++++++
 stack/l2cap/l2c_utils.c |  6 ++++++
 6 files changed, 43 insertions(+)

diff --git a/Android.mk b/Android.mk
index 29760e4..8ab15ee 100644
--- a/Android.mk
+++ b/Android.mk
@@ -41,6 +41,11 @@ bluetooth_CFLAGS += \
   -UNDEBUG \
   -DLOG_NDEBUG=1
 
+# Added for compatability with LE/GATT disabled
+bluetooth_CFLAGS += \
+    -Wno-unused-variable \
+    -Wno-unused-function
+
 bluetooth_CONLYFLAGS += -std=c99
 bluetooth_CPPFLAGS :=
 
diff --git a/bta/dm/bta_dm_act.c b/bta/dm/bta_dm_act.c
index 2fa9d01..b320916 100644
--- a/bta/dm/bta_dm_act.c
+++ b/bta/dm/bta_dm_act.c
@@ -383,7 +383,9 @@ static void bta_dm_sys_hw_cback( tBTA_SYS_HW_EVT status )
 
         /* hw is ready, go on with BTA DM initialization */
         alarm_free(bta_dm_search_cb.search_timer);
+#if (defined(BTA_GATT_INCLUDED) && (BTA_GATT_INCLUDED == TRUE))
         alarm_free(bta_dm_search_cb.gatt_close_timer);
+#endif
         memset(&bta_dm_search_cb, 0, sizeof(bta_dm_search_cb));
         /*
          * TODO: Should alarm_free() the bta_dm_search_cb timers during
@@ -391,8 +393,10 @@ static void bta_dm_sys_hw_cback( tBTA_SYS_HW_EVT status )
          */
         bta_dm_search_cb.search_timer =
           alarm_new("bta_dm_search.search_timer");
+#if (defined(BTA_GATT_INCLUDED) && (BTA_GATT_INCLUDED == TRUE))
         bta_dm_search_cb.gatt_close_timer =
           alarm_new("bta_dm_search.gatt_close_timer");
+#endif
 
         memset(&bta_dm_conn_srvcs, 0, sizeof(bta_dm_conn_srvcs));
         memset(&bta_dm_di_cb, 0, sizeof(tBTA_DM_DI_CB));
diff --git a/btif/src/btif_dm.c b/btif/src/btif_dm.c
index c47909f..c297f5b 100644
--- a/btif/src/btif_dm.c
+++ b/btif/src/btif_dm.c
@@ -2681,8 +2681,10 @@ bt_status_t btif_dm_get_remote_services_by_transport(bt_bdaddr_t *remote_addr, c
     mask_ext.p_uuid = NULL;
     mask_ext.srvc_mask = BTA_ALL_SERVICE_MASK;
 
+#if ((defined BLE_INCLUDED) && (BLE_INCLUDED == TRUE))
     BTA_DmDiscoverByTransport(remote_addr->address, &mask_ext,
                    bte_dm_search_services_evt, TRUE, transport);
+#endif
 
     return BT_STATUS_SUCCESS;
 }
@@ -2921,6 +2923,7 @@ BOOLEAN btif_dm_get_smp_config(tBTE_APPL_CFG* p_cfg) {
     strncpy(conf, recv, 64);
     conf[63] = 0; // null terminate
 
+#if (defined(BLE_INCLUDED) && (BLE_INCLUDED == TRUE))
     if ((pch = strtok(conf, ",")) != NULL)
         p_cfg->ble_auth_req = (UINT8) strtoul(pch, &endptr, 16);
     else
@@ -2945,6 +2948,7 @@ BOOLEAN btif_dm_get_smp_config(tBTE_APPL_CFG* p_cfg) {
         p_cfg->ble_max_key_size =  (UINT8) strtoul(pch, &endptr, 16);
     else
         return FALSE;
+#endif
 
     return TRUE;
 }
diff --git a/btif/src/btif_storage.c b/btif/src/btif_storage.c
index 82d377d..1541edd 100644
--- a/btif/src/btif_storage.c
+++ b/btif/src/btif_storage.c
@@ -502,6 +502,7 @@ static void btif_read_le_key(const uint8_t key_type, const size_t key_len, bt_bd
     char buffer[100];
     memset(buffer, 0, sizeof(buffer));
 
+#if ((defined BLE_INCLUDED) && (BLE_INCLUDED == TRUE))
     if (btif_storage_get_ble_bonding_key(&bd_addr, key_type, buffer, key_len) == BT_STATUS_SUCCESS)
     {
         if (add_key)
@@ -523,6 +524,7 @@ static void btif_read_le_key(const uint8_t key_type, const size_t key_len, bt_bd
 
         *key_found = true;
     }
+#endif
 }
 
 /*******************************************************************************
diff --git a/stack/l2cap/l2c_csm.c b/stack/l2cap/l2c_csm.c
index dedffc8..1f939d5 100644
--- a/stack/l2cap/l2c_csm.c
+++ b/stack/l2cap/l2c_csm.c
@@ -172,8 +172,10 @@ static void l2c_csm_closed (tL2C_CCB *p_ccb, UINT16 event, void *p_data)
         if (p_ccb->p_lcb->transport == BT_TRANSPORT_LE)
         {
             p_ccb->chnl_state = CST_ORIG_W4_SEC_COMP;
+#if ((defined BLE_INCLUDED) && (BLE_INCLUDED == TRUE))
             l2ble_sec_access_req(p_ccb->p_lcb->remote_bd_addr, p_ccb->p_rcb->psm, TRUE,
                     &l2c_link_sec_comp, p_ccb);
+#endif
         }
         else
         {
@@ -198,8 +200,10 @@ static void l2c_csm_closed (tL2C_CCB *p_ccb, UINT16 event, void *p_data)
         if (p_ccb->p_lcb->transport == BT_TRANSPORT_LE)
         {
             p_ccb->chnl_state = CST_ORIG_W4_SEC_COMP;
+#if ((defined BLE_INCLUDED) && (BLE_INCLUDED == TRUE))
             l2ble_sec_access_req(p_ccb->p_lcb->remote_bd_addr, p_ccb->p_rcb->psm, TRUE,
                     &l2c_link_sec_comp, p_ccb);
+#endif
         }
         else
         {
@@ -261,8 +265,10 @@ Event uninit_use_in_call: Using uninitialized value "settings.min" in call to fu
         if (p_ccb->p_lcb->transport == BT_TRANSPORT_LE)
         {
             p_ccb->chnl_state = CST_TERM_W4_SEC_COMP;
+#if ((defined BLE_INCLUDED) && (BLE_INCLUDED == TRUE))
              l2ble_sec_access_req(p_ccb->p_lcb->remote_bd_addr, p_ccb->p_rcb->psm, FALSE,
                     &l2c_link_sec_comp, p_ccb);
+#endif
         }
         else
         {
@@ -360,8 +366,10 @@ static void l2c_csm_orig_w4_sec_comp (tL2C_CCB *p_ccb, UINT16 event, void *p_dat
     case L2CEVT_LP_CONNECT_CFM:                     /* Link came up         */
         if (p_ccb->p_lcb->transport == BT_TRANSPORT_LE)
         {
+#if ((defined BLE_INCLUDED) && (BLE_INCLUDED == TRUE))
              l2ble_sec_access_req(p_ccb->p_lcb->remote_bd_addr, p_ccb->p_rcb->psm, FALSE,
                     &l2c_link_sec_comp, p_ccb);
+#endif
         }
         else
         {
@@ -375,11 +383,13 @@ static void l2c_csm_orig_w4_sec_comp (tL2C_CCB *p_ccb, UINT16 event, void *p_dat
         p_ccb->chnl_state = CST_W4_L2CAP_CONNECT_RSP;
         if (p_ccb->p_lcb->transport == BT_TRANSPORT_LE)
         {
+#if ((defined BLE_INCLUDED) && (BLE_INCLUDED == TRUE))
             alarm_set_on_queue(p_ccb->l2c_ccb_timer,
                                L2CAP_CHNL_CONNECT_TIMEOUT_MS,
                                l2c_ccb_timer_timeout, p_ccb,
                                btu_general_alarm_queue);
             l2cble_credit_based_conn_req (p_ccb);          /* Start Connection     */
+#endif
         }
         else
         {
@@ -514,9 +524,11 @@ static void l2c_csm_term_w4_sec_comp (tL2C_CCB *p_ccb, UINT16 event, void *p_dat
         }
         else
         {
+#if ((defined BLE_INCLUDED) && (BLE_INCLUDED == TRUE))
             if (p_ccb->p_lcb->transport == BT_TRANSPORT_LE)
                 l2cu_reject_ble_connection(p_ccb->p_lcb, p_ccb->remote_id, L2CAP_LE_INSUFFICIENT_AUTHENTICATION);
             else
+#endif
                 l2cu_send_peer_connect_rsp (p_ccb, L2CAP_CONN_SECURITY_BLOCK, 0);
             l2cu_release_ccb (p_ccb);
         }
@@ -723,6 +735,7 @@ static void l2c_csm_w4_l2ca_connect_rsp (tL2C_CCB *p_ccb, UINT16 event, void *p_
         p_ci = (tL2C_CONN_INFO *)p_data;
         if (p_ccb->p_lcb->transport == BT_TRANSPORT_LE)
         {
+#if ((defined BLE_INCLUDED) && (BLE_INCLUDED == TRUE))
             /* Result should be OK or Reject */
             if ((!p_ci) || (p_ci->l2cap_result == L2CAP_CONN_OK))
             {
@@ -735,6 +748,7 @@ static void l2c_csm_w4_l2ca_connect_rsp (tL2C_CCB *p_ccb, UINT16 event, void *p_
                 l2cble_credit_based_conn_res (p_ccb, p_ci->l2cap_result);
                 l2cu_release_ccb (p_ccb);
             }
+#endif
         }
         else
         {
@@ -762,9 +776,11 @@ static void l2c_csm_w4_l2ca_connect_rsp (tL2C_CCB *p_ccb, UINT16 event, void *p_
 
     case L2CEVT_L2CA_CONNECT_RSP_NEG:
         p_ci = (tL2C_CONN_INFO *)p_data;
+#if ((defined BLE_INCLUDED) && (BLE_INCLUDED == TRUE))
         if (p_ccb->p_lcb->transport == BT_TRANSPORT_LE)
             l2cble_credit_based_conn_res (p_ccb, p_ci->l2cap_result);
         else
+#endif
             l2cu_send_peer_connect_rsp (p_ccb, p_ci->l2cap_result, p_ci->l2cap_status);
         l2cu_release_ccb (p_ccb);
         break;
@@ -1185,9 +1201,11 @@ static void l2c_csm_open (tL2C_CCB *p_ccb, UINT16 event, void *p_data)
             }
         }
 
+#if ((defined BLE_INCLUDED) && (BLE_INCLUDED == TRUE))
         if (p_ccb->p_lcb->transport == BT_TRANSPORT_LE)
             l2cble_send_peer_disc_req (p_ccb);
         else
+#endif
             l2cu_send_peer_disc_req (p_ccb);
 
         p_ccb->chnl_state = CST_W4_L2CAP_DISCONNECT_RSP;
@@ -1226,7 +1244,9 @@ static void l2c_csm_open (tL2C_CCB *p_ccb, UINT16 event, void *p_data)
     case L2CEVT_L2CA_SEND_FLOW_CONTROL_CREDIT:
         L2CAP_TRACE_DEBUG("%s Sending credit",__func__);
         credit = (UINT16*)p_data;
+#if ((defined BLE_INCLUDED) && (BLE_INCLUDED == TRUE))
         l2cble_send_flow_control_credit(p_ccb, *credit);
+#endif
         break;
 
     case L2CEVT_L2CAP_RECV_FLOW_CONTROL_CREDIT:
@@ -1234,10 +1254,12 @@ static void l2c_csm_open (tL2C_CCB *p_ccb, UINT16 event, void *p_data)
         L2CAP_TRACE_DEBUG("%s Credits received %d",__func__, *credit);
         if((p_ccb->peer_conn_cfg.credits + *credit) > L2CAP_LE_MAX_CREDIT)
         {
+#if ((defined BLE_INCLUDED) && (BLE_INCLUDED == TRUE))
             /* we have received credits more than max coc credits,
              * so disconnecting the Le Coc Channel
              */
             l2cble_send_peer_disc_req (p_ccb);
+#endif
         }
         else
         {
diff --git a/stack/l2cap/l2c_utils.c b/stack/l2cap/l2c_utils.c
index 58ac4b0..d956f1e 100644
--- a/stack/l2cap/l2c_utils.c
+++ b/stack/l2cap/l2c_utils.c
@@ -253,6 +253,7 @@ void l2cu_release_lcb (tL2C_LCB *p_lcb)
         (*p_cb) (L2CAP_PING_RESULT_NO_LINK);
     }
 
+#if ((defined BLE_INCLUDED) && (BLE_INCLUDED == TRUE))
     /* Check and release all the LE COC connections waiting for security */
     if (p_lcb->le_sec_pending_q)
     {
@@ -266,6 +267,7 @@ void l2cu_release_lcb (tL2C_LCB *p_lcb)
         fixed_queue_free(p_lcb->le_sec_pending_q, NULL);
         p_lcb->le_sec_pending_q = NULL;
     }
+#endif
 }
 
 
@@ -1829,6 +1831,7 @@ tL2C_RCB *l2cu_allocate_rcb (UINT16 psm)
 *******************************************************************************/
 tL2C_RCB *l2cu_allocate_ble_rcb (UINT16 psm)
 {
+#if ((defined BLE_INCLUDED) && (BLE_INCLUDED == TRUE))
     tL2C_RCB    *p_rcb = &l2cb.ble_rcb_pool[0];
     UINT16      xx;
 
@@ -1844,6 +1847,7 @@ tL2C_RCB *l2cu_allocate_ble_rcb (UINT16 psm)
             return (p_rcb);
         }
     }
+#endif
 
     /* If here, no free RCB found */
     return (NULL);
@@ -1934,6 +1938,7 @@ tL2C_RCB *l2cu_find_rcb_by_psm (UINT16 psm)
 *******************************************************************************/
 tL2C_RCB *l2cu_find_ble_rcb_by_psm (UINT16 psm)
 {
+#if ((defined BLE_INCLUDED) && (BLE_INCLUDED == TRUE))
     tL2C_RCB    *p_rcb = &l2cb.ble_rcb_pool[0];
     UINT16      xx;
 
@@ -1942,6 +1947,7 @@ tL2C_RCB *l2cu_find_ble_rcb_by_psm (UINT16 psm)
         if ((p_rcb->in_use) && (p_rcb->psm == psm))
             return (p_rcb);
     }
+#endif
 
     /* If here, no match found */
     return (NULL);
-- 
2.7.4

